src/protobuf.zig

```diff
@@ -586,7 +586,7 @@ fn deinit_field(result: anytype, comptime field_name: []const u8, comptime ftype
             }
         },
         .List => |list_type| {
-            if (list_type == .SubMessage or list_type == .String) {
+            if (list_type == .SubMessage or list_type == .String or list_type == .Bytes) {
```

Comment:
Let's use switch here so we have the exhaustive check at compile time and avoid this kind of error in the future.

```patch
diff --git a/src/protobuf.zig b/src/protobuf.zig
index dbe0c48..ac7e811 100644
--- a/src/protobuf.zig
+++ b/src/protobuf.zig
@@ -586,10 +586,13 @@ fn deinit_field(result: anytype, comptime field_name: []const u8, comptime ftype
             }
         },
         .List => |list_type| {
-            if (list_type == .SubMessage or list_type == .String or list_type == .Bytes) {
-                for (@field(result, field_name).items) |item| {
-                    item.deinit();
-                }
+            switch (list_type) {
+                .SubMessage, .String, .Bytes => {
+                    for (@field(result, field_name).items) |item| {
+                        item.deinit();
+                    }
+                },
+                .Varint, .FixedInt => {},
             }
             @field(result, field_name).deinit();
         },
```

Reply:
Ok, I'll change it. Thanks for the improvement. I'm currently taking my first baby steps with zig, so I'm grateful for each learning along the way :)